File: src/test/java/com/github/kayjamlang/tests/expressions/NamedExpressionTest.java
package com.github.kayjamlang.tests.expressions;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.containers.Container;
import com.github.kayjamlang.core.expressions.NamedExpression;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class NamedExpressionTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("test{}"));
    }

    @Test
    public void test() throws Exception {
        Expression expression = parser.readExpression();

        assertNotNull(expression);
        assertSame(NamedExpression.class, expression.getClass());

        NamedExpression namedContainer = (NamedExpression) expression;
        assertSame(Container.class, namedContainer.expression.getClass());
    }
}

File: src/test/scala/com/github/kayjamlang/tests/expressions/NamedExpressionTest.scala
package com.github.kayjamlang.tests.expressions

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.containers.Container
import com.github.kayjamlang.core.expressions.NamedExpression
import org.junit.Assert._
import org.junit.{BeforeClass, Test}

object NamedExpressionTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("test{}"))
  }
}

class NamedExpressionTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = NamedExpressionTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[NamedExpression], expression getClass)
    val namedContainer = expression.asInstanceOf[NamedExpression]
    assertSame(classOf[Container], namedContainer.expression getClass)
  }
}